kompari/src/imageutils.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2025 the Kompari Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::Image;
+use std::io::Cursor;
+use std::path::Path;
+
+pub fn load_image(path: &Path) -> crate::Result<Image> {
+    if !path.is_file() {
+        return Err(crate::Error::FileNotFound(path.to_path_buf()));
+    }
+    Ok(image::ImageReader::open(path)?.decode()?.into_rgba8())
+}
+
+#[cfg(feature = "oxipng")]
+pub fn optimize_png(data: Vec<u8>) -> Vec<u8> {
+    oxipng::optimize_from_memory(&data[..], &oxipng::Options::default()).unwrap_or(data)
```

Comment:
I think different use cases have different needs for the options; for embedding into the report, the default (effectively preset 2) is probably fine, but e.g. preset 5 might be better for saving to the file system.

Reply:
Introduced two size optimization levels.